packages/amplify-cli/src/extensions/amplify-helpers/resource-status-data.ts

```diff
@@ -217,8 +217,9 @@ export function getResourcesToBeCreated(amplifyMeta, currentAmplifyMeta, categor
           (!amplifyMeta[dependsOnCategory][dependsOnResourcename]?.lastPushTimeStamp ||
             !currentAmplifyMeta[dependsOnCategory] ||
             !currentAmplifyMeta[dependsOnCategory][dependsOnResourcename]) &&
-          amplifyMeta[dependsOnCategory][dependsOnResourcename].serviceType !== 'imported' &&
-          !resources.includes(amplifyMeta[dependsOnCategory][dependsOnResourcename])
+          amplifyMeta[dependsOnCategory][dependsOnResourcename]?.serviceType !== 'imported' &&
+          !resources.includes(amplifyMeta[dependsOnCategory][dependsOnResourcename]) &&
+          amplifyMeta[dependsOnCategory][dependsOnResourcename]
```

Comment:
Can you move this up two lines (so that it's before the `serviceType` check). That way, the logic should short circuit a little faster.

Reply:
Yathi also moved around these, took those changes.